### DISCUSSION TITLE
Downgrade failure should be recompiled

### DISCUSSION THREAD

**Discussion by t812206236** [Homebrew/discussions#1597] [Everyday usage] [answered]
I want to downgrade a software called qemu. I found its historical version in homebrew-core. When I want to use the command to install it, it told me that it failed to get the manifest from the `ghcr.io/v2/homebrew/core` website. 

I hesitated for a long time to figure out the possibility. It may be because this website does not provide the manifest of the old version, which causes the acquisition to fail.

I remember that it will automatically compile if it fails to obtain it before. Why the current performance is to withdraw directly？

Finally, I used the parameter `--build-from-source` to solve the problem, but I think this function should be executed automatically.

**Comment by SMillerDev** [comment, marked as answer]
Homebrew doesn't build from source by default because it goes wrong more often than not. https://docs.brew.sh/Manpage#extract---version---force-formula-tap is the command that will allow you to extract older versions of formula
